### PR TITLE
Add automatic detection of k8s prior 1.21

### PR DESCRIPTION
**Pull request by juldrixx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | -
| License         | Apache 2.0

### What's in this PR?

Detection of Kubernetes version to revert some changes if the version is prior 1.21.

### Why?

To be compatible with version prior 1.21 of Kubernetes.

### Checklist

- [ ] Implementation tested
- [x] Error handling code meets the [guideline](docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)
- [x] Append changelog with changes

### CODE REVIEW

File: config/crd/bases/nifi.konpyutaika.com_nifinodegroupautoscalers.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.8.0
+    controller-gen.kubebuilder.io/version: v0.9.0
```
Comment (cannonpalms) [inline config/crd/bases/nifi.konpyutaika.com_nifinodegroupautoscalers.yaml:6]: Did you intend for changes to this file to be included in this PR, or is this a result of an errant `make`? I've done the latter several times.

**Comment by cannonpalms** [comment]
LGTM outside of my question about generated files.

File: config/crd/bases/nifi.konpyutaika.com_nifinodegroupautoscalers.yaml (hunk 1 above)
Comment (michael81877) [reply config/crd/bases/nifi.konpyutaika.com_nifinodegroupautoscalers.yaml:6]: Yes, this is an oversight following #120 . kubebuilder was upgraded in operator-sdk v1.22.1, so it should be included. Thank you for double checking! 👍

File: pkg/util/util.go [hunk 2]
```diff
@@ -283,3 +285,23 @@ func GetRequeueInterval(interval int, offset int) time.Duration {
 	duration = Max(duration, rand.Intn(5)+1) // make sure duration does not go zero for very large offsets
 	return time.Duration(duration) * time.Second
 }
+
+func IsK8sPrior1_21() bool {
```
[the remark this answers is not in the document]
Comment (juldrixx) [reply pkg/util/util.go:289]: yes, I thought about it but I was lazy. I will take your code , thanks 😎

[the remark this answers is not in the document]
Comment (juldrixx) [reply pkg/util/util.go:289]: I made some modifications to make it work 😉

File: config/crd/bases/nifi.konpyutaika.com_nifinodegroupautoscalers.yaml (hunk 1 above)
Comment (juldrixx) [reply config/crd/bases/nifi.konpyutaika.com_nifinodegroupautoscalers.yaml:6]: I did not made the change, the change was done automatically by something when I did `make generate manifests`.

File: pkg/util/util.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (michael81877) [reply pkg/util/util.go:289]: Cheers! :)

**Comment by michael81877** [comment]
LGTM as well

**Review by erdrix — APPROVED** [review]

**Pull request merged**